mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -298,11 +298,7 @@ class LinearRegressionModel private[ml] (
    */
   // TODO: decide on a good name before exposing to public API
   private[regression] def evaluate(dataset: DataFrame): LinearRegressionSummary = {
-    val t = udf { features: Vector => predict(features) }
-    val predictionAndObservations = dataset
-      .select(col($(labelCol)), t(col($(featuresCol))).as($(predictionCol)))
-
-    new LinearRegressionSummary(predictionAndObservations, $(predictionCol), $(labelCol))
+    new LinearRegressionSummary(transform(dataset), $(predictionCol), $(labelCol))
```

Comment:
No, it should permit columns to be turned off by setting them to empty strings.  That's useful for classification (with multiple output columns).  It may not be useful for regression currently, but will be when we add other outputs (like variance estimates).